src/awkward/_v2/contents/emptyarray.py

```diff
@@ -205,6 +205,13 @@ def _reduce_next(
     def _validityerror(self, path):
         return ""
 
+    def _rpad(self, target, axis, depth, clip):
+        posaxis = self.axis_wrap_if_negative(axis)
+        if posaxis != depth:
+            raise ValueError("axis exceeds the depth of this array")
```

Comment:
Errors in which `axis=XYZ` and `XYZ` is not a valid choice are handled all over the place in the codebase, but they mean the same thing to the user. I've been trying to consolidate their messages, so that they get to be more immediately recognizable, in part by using NumPy's `np.AxisError`, and by using the same message text.

```python
raise np.AxisError("axis={0} exceeds the depth of this array ({1})".format(axis, depth))
```

The `axis` is the `XYZ` that the user specifies (may be negative) and `depth` has a consistent meaning in all of these functions (fortunately!). In principle, one might think this calls for handling all axis errors in a subroutine, but then the last line number of the stack trace would be in that subroutine; to debug it, we'd have to remember to go one level up, and that's a little inconvenient. Also, _calling_ such a subroutine would be something we'd have to do consistently everywhere, and I don't see how it's any harder to just use consistent text. (The only argument would be if we planned on ever changing that text—I think we can safely ignore that possibility.)

So, please do change the exception class and message for this and any other "bad axis" exceptions.

Reply:
This is done in the latest commits.